File: projects/lostengine/lost/lua/lost/io/rg/Blend.lua
module("lost.io", package.seeall)

require("lost.io.Loader")

-- calls the glEnable/glDisable command with GL_BLEND
function Loader:Blend(definition)
  local enable = definition["enable"]
  if enable == nil then
    error("enable is required for rg:Blend")
  end
  local result = lost.rg.Blend.create(enable)
  
  local active = definition["active"]
  local name = definition["name"]
  if active ~= nil then
    result.active = active
  end
  if name ~= nil then
    result.name = name
  end
  return result
end

File: projects/lostengine/lost/rg/Blend.cpp
#include "lost/rg/Blend.h"
#include "lost/gl/Context.h"

namespace lost
{
  namespace rg
  {
    
    Blend::Blend(bool inEnable)
    :enable(inEnable)
    {
      name = "blend";
    }
    
    BlendPtr Blend::create(bool inEnable)
    {
      return BlendPtr(new Blend(inEnable));
    }
    
    void Blend::process(gl::ContextPtr& ctx)
    {
      if(!active) return;
      ctx->blend(enable);
    }
    
  }
}